src/platform/packages/shared/response-ops/rule_params/custom_threshold/v1.ts

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+import { schema } from '@kbn/config-schema';
+import { COMPARATORS } from '@kbn/alerting-comparators';
+import { dataViewSpecSchema } from '../common';
+import { oneOfLiterals, validateKQLStringFilter, LEGACY_COMPARATORS } from '../common/utils';
+
+const allowedAggregators = [
```

Comment:
I see this part has changed:

```
  const allowedAggregators = Object.values(Aggregators);
  allowedAggregators.splice(Object.values(Aggregators).indexOf(Aggregators.COUNT), 1);
```

Do we still have the `Aggregators` types somewhere? I am trying to remember why we had two list and excluded one from the allowed list.

Reply:
So, it seems we have this type still in common ([Aggregator](https://github.com/elastic/kibana/blob/main/x-pack/solutions/observability/plugins/observability/common/custom_threshold_rule/types.ts#L25)), and we use it in a lot of places.

I like the idea of having this aggregator shared everywhere (i.e., moving the common aggregator type here), but then this will increase the scope of this PR. (There is no issue in using this package both in FE and server-side, right?)

@cnasikas thoughts?

Reply:
We decided not to change the way solutions use the types and the schemas due to a lack of domain knowledge. Also, some of the schemas are made with `io-ts` and we need the schemas to be in `kbn-config`. When the task is finished, we plan to inform you that you can use the package for types/schemas. It is up to the solutions to make the changes and use the types/schemas from the package if they want.

Reply:
Here is an idea that minimizes the number of changes, let me know what you think:

1. Moving [Aggregators](https://github.com/elastic/kibana/blob/main/x-pack/solutions/observability/plugins/observability/common/custom_threshold_rule/types.ts#L25) enum from this package.
2. Use it as before here:
    ```
    const allowedAggregators = Object.values(Aggregators);
    allowedAggregators.splice(Object.values(Aggregators).indexOf(Aggregators.COUNT), 1);
    ```
3. Instead of `aggType: schema.literal('count'),`, use
    ```
    aggType: schema.literal(Aggregators.COUNT)
    ```
    to only rely on the enum.
4. Import Aggregators from here in the common, so we don't need to change anything else.

Would that work?

Reply:
> (There is no issue using this package on both the FE and server sides, right?)

I thought there was an issue using this in the FE since `rule_params` uses the `config-schema` which relies on some `library(hapi`?) which is not usable on the front end. Maybe @cnasikas can confirm?

If so we could move this to a different `rule_params_constants` package and then import it like you say.

As for the `allowedAggregators` values + splice logic, I don't have a better alternative :(

Reply:
The main point here is to use Aggregators Enum both here and observability > common. So, if there is another package that can be used for this purpose, let's use that 👍🏻

Reply:
> I thought there was an issue using this in the FE since rule_params uses the config-schema which relies on some library(hapi?) which is not usable on the front end.

Goor point about not using config schema on the FE, that's matches my understanding as well.

Reply:
> So, if there is another package that can be used for this purpose, let's use that

Not yet :D but I think that would make sense, no need to get the whole `rule_params` package on the ride.

> The main point here is to use Aggregators Enum both here and observability > common.

I fully agree with having a single point of truth for these kind of stuff

Reply:
As @adcoelho said, the package is intended to be used only on the server side. We cannot import stuff from this package in the `common` or the `public` folder (types can imported anywhere). Also, I would prefer to have a copy of our literals because a shared `enum` (between packages) may evolve in the future and accidentally introduce a breaking change to our schema. Lastly, it seems that the `enum` is used in more places than the rule type, which is something the ResponseOps team cannot own in the long run.

Reply:
> it seems that the enum is used in more places than the rule type, which is something the ResponseOps team cannot own in the long run.

Where else is it used?

Reply:
Searching in VSCode reveals a lot of places in the o11y plugin, like the AI assistant, for example.

Reply:
I checked again, and I don't see a usage other than related to the custom threshold rule. The AI assistant usage is also related to creating a custom threshold rule.

Is there another package related to rule types from which we can expose the Aggregators enum and use it in both places?

Reply:
I think the confusion comes from considering this enum separate from being a rule param. For us, there is no difference between this enum and the Aggregators that we use in the rule, and defining it as an enum helps us to keep the value in sync in all the usages.

> Who will own the centralized package Aggregators enum?

This would be the same as this current package for exposing rule parameters, we just separate it to use it on FE as well due to the bundle limitation.

It would be good to have this improvement if we have a package that we can expose this enum, but I don't see it as a blocker, so I'll approve this PR.

Reply:
Thanks Maryam! I get your points. Let's talk about it offline.

Reply:
I created an issue for a `rule_params_constants` package. https://github.com/elastic/kibana/issues/209547